File: backend/src/main/java/com/creditscoring/entity/CreditRequest.java
package com.creditscoring.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "credit_requests")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreditRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @Column(name = "request_number", unique = true, nullable = false, length = 50)
    private String requestNumber;
    
    @Column(name = "amount_requested", nullable = false, precision = 15, scale = 2)
    private BigDecimal amountRequested;
    
    private String purpose;
    
    @Column(name = "employment_status", length = 50)
    private String employmentStatus;
    
    @Column(name = "monthly_income", precision = 15, scale = 2)
    private BigDecimal monthlyIncome;
    
    @Column(name = "employment_duration_months")
    private Integer employmentDurationMonths;
    
    @Column(name = "current_debt", precision = 15, scale = 2)
    private BigDecimal currentDebt = BigDecimal.ZERO;
    
    @Column(name = "credit_history_months")
    private Integer creditHistoryMonths;
    
    private Integer age;
    
    @Column(name = "marital_status", length = 20)
    private String maritalStatus;
    
    private Integer dependents = 0;
    
    @Column(name = "property_owned")
    private Boolean propertyOwned = false;
    
    @Column(name = "education_level", length = 50)
    private String educationLevel;
    
    @Column(name = "ai_score", precision = 5, scale = 2)
    private BigDecimal aiScore;
    
    @Column(name = "ai_confidence", precision = 5, scale = 2)
    private BigDecimal aiConfidence;
    
    @Column(name = "final_score", precision = 5, scale = 2)
    private BigDecimal finalScore;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RequestStatus status = RequestStatus.PENDING;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "agent_id")
    private User agent;
    
    @Column(name = "validated_at")
    private LocalDateTime validatedAt;
    
    @Column(name = "rejection_reason", columnDefinition = "TEXT")
    private String rejectionReason;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
    
    public enum RequestStatus {
        PENDING, PROCESSING, APPROVED, REJECTED, TREATED
    }
}

File: frontend/lib/models/credit_request.dart
class CreditRequest {
  final int id;
  final String requestNumber;
  final double amountRequested;
  final String? purpose;
  final String? employmentStatus;
  final double? monthlyIncome;
  final int? employmentDurationMonths;
  final double? currentDebt;
  final int? creditHistoryMonths;
  final int? age;
  final String? maritalStatus;
  final int? dependents;
  final bool? propertyOwned;
  final String? educationLevel;
  final double? aiScore;
  final double? aiConfidence;
  final double? finalScore;
  final String status;
  final int? agentId;
  final DateTime? validatedAt;
  final String? rejectionReason;
  final DateTime? createdAt;
  final UserInfo? user;

  CreditRequest({
    required this.id,
    required this.requestNumber,
    required this.amountRequested,
    this.purpose,
    this.employmentStatus,
    this.monthlyIncome,
    this.employmentDurationMonths,
    this.currentDebt,
    this.creditHistoryMonths,
    this.age,
    this.maritalStatus,
    this.dependents,
    this.propertyOwned,
    this.educationLevel,
    this.aiScore,
    this.aiConfidence,
    this.finalScore,
    required this.status,
    this.agentId,
    this.validatedAt,
    this.rejectionReason,
    this.createdAt,
    this.user,
  });

  factory CreditRequest.fromJson(Map<String, dynamic> json) {
    return CreditRequest(
      id: json['id'],
      requestNumber: json['requestNumber'],
      amountRequested: (json['amountRequested'] as num).toDouble(),
      purpose: json['purpose'],
      employmentStatus: json['employmentStatus'],
      monthlyIncome: json['monthlyIncome'] != null ? (json['monthlyIncome'] as num).toDouble() : null,
      employmentDurationMonths: json['employmentDurationMonths'],
      currentDebt: json['currentDebt'] != null ? (json['currentDebt'] as num).toDouble() : null,
      creditHistoryMonths: json['creditHistoryMonths'],
      age: json['age'],
      maritalStatus: json['maritalStatus'],
      dependents: json['dependents'],
      propertyOwned: json['propertyOwned'],
      educationLevel: json['educationLevel'],
      aiScore: json['aiScore'] != null ? (json['aiScore'] as num).toDouble() : null,
      aiConfidence: json['aiConfidence'] != null ? (json['aiConfidence'] as num).toDouble() : null,
      finalScore: json['finalScore'] != null ? (json['finalScore'] as num).toDouble() : null,
      status: json['status'],
      agentId: json['agentId'],
      validatedAt: json['validatedAt'] != null ? DateTime.parse(json['validatedAt']) : null,
      rejectionReason: json['rejectionReason'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      user: json['user'] != null ? UserInfo.fromJson(json['user']) : null,
    );
  }

  bool get isPending => status == 'PENDING';
  bool get isTreated => status == 'TREATED';
  bool get isApproved => status == 'APPROVED';
  bool get isRejected => status == 'REJECTED';
}

class UserInfo {
  final int id;
  final String? email;
  final String? firstName;
  final String? lastName;

  UserInfo({
    required this.id,
    this.email,
    this.firstName,
    this.lastName,
  });

  factory UserInfo.fromJson(Map<String, dynamic> json) {
    return UserInfo(
      id: json['id'],
      email: json['email'],
      firstName: json['firstName'],
      lastName: json['lastName'],
    );
  }

  String get fullName => '${firstName ?? ''} ${lastName ?? ''}'.trim();
}

